internal/image/fixtures/schemazstdselection.json

```diff
@@ -0,0 +1,27 @@
+{
+    "schemaVersion": 2,
+    "mediaType": "application/vnd.oci.image.index.v1+json",
+    "manifests": [
+        {
+            "mediaType": "application/vnd.oci.image.manifest.v1+json",
+            "digest": "sha256:ac9bbf71dacfad2bfbe48f7d46ca279b1b3b351a6c814db760030015167405b8",
+            "size": 772,
+            "annotations": {
+                "io.github.containers.compression.zstd": "true"
```

Comment:
We need more tests.  In this particular case, the first image would always be picked - with or without the code changes.  The crux of the matter is to walk the entire list and pick the ZSTD matching one.

In the end, the gzip compressed ones must always be listed before the zstd ones to remain backwards compatible.

Reply:
Added another test where gzip is the first listed image , and added more images in unordered manner. Could you suggest  some more tests if i missed anything.

Reply:
There could be a manifest with multiple matching gzip-compressed and multiple matching zstd-compressed images.  We should test this case and make sure to select the expected one(s).  Either the first gzip or the first zstd one - depending on what we're looking for.

Reply:
(Note to self: I didn’t look at the test coverage.)